### PR TITLE
feat: Add PSRule analysis for WAF Reliability Pillar for module defaults - V2

### PR DESCRIPTION
**Pull request by jtracey93**
## Description

Add PSRule analysis for WAF Reliability Pillar for module defaults

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.res.analysis-services.server](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.analysis-services.server.yml)
[avm.res.api-management.service](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.api-management.service.yml)
[avm.res.app-configuration.configuration-store](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.app-configuration.configuration-store.yml)
[avm.res.app.container-app](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.app.container-app.yml)
[avm.res.app.managed-environment](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.app.managed-environment.yml)
[avm.res.automation.automation-account](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.automation.automation-account.yml)
[avm.res.batch.batch-account](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.batch.batch-account.yml)
[avm.res.cache.redis](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.cache.redis.yml)
[avm.res.cdn.profile](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.cdn.profile.yml)
[avm.res.cognitive-services.account](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.cognitive-services.account.yml)
[avm.res.compute.availability-set](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.compute.availability-set.yml)
[avm.res.compute.disk-encryption-set](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.compute.disk-encryption-set.yml)
[avm.res.compute.disk](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.compute.disk.yml)
[avm.res.compute.gallery](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.compute.gallery.yml)
[avm.res.compute.image](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.compute.image.yml)
[avm.res.compute.proximity-placement-group](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.compute.proximity-placement-group.yml)
[avm.res.compute.ssh-public-key](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.compute.ssh-public-key.yml)
[avm.res.compute.virtual-machine-scale-set](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine-scale-set.yml)
[avm.res.compute.virtual-machine](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine.yml)
[avm.res.consumption.budget](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.consumption.budget.yml)
[avm.res.container-registry.registry](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.container-registry.registry.yml)
[avm.res.container-service.managed-cluster](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.container-service.managed-cluster.yml)
[avm.res.data-factory.factory](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.data-factory.factory.yml)      
[avm.res.data-protection.backup-vault](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.data-protection.backup-vault.yml)
[avm.res.databricks.access-connector](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.databricks.access-connector.yml)
[avm.res.databricks.workspace](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.databricks.workspace.yml)      
[avm.res.db-for-my-sql.flexible-server](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.db-for-my-sql.flexible-server.yml)
[avm.res.db-for-postgre-sql.flexible-server](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.db-for-postgre-sql.flexible-server.yml)
[avm.res.desktop-virtualization.application-group](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.application-group.yml)
[avm.res.desktop-virtualization.host-pool](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.host-pool.yml)
[avm.res.desktop-virtualization.scaling-plan](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.scaling-plan.yml)
[avm.res.desktop-virtualization.workspace](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.workspace.yml)
[avm.res.dev-test-lab.lab](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.dev-test-lab.lab.yml)
[avm.res.document-db.database-account](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.document-db.database-account.yml)
[avm.res.event-grid.domain](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.event-grid.domain.yml)
[avm.res.event-grid.namespace](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.event-grid.namespace.yml)      
[avm.res.event-grid.system-topic](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.event-grid.system-topic.yml)
[avm.res.event-grid.topic](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.event-grid.topic.yml)
[avm.res.event-hub.namespace](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.event-hub.namespace.yml)
[avm.res.health-bot.health-bot](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.health-bot.health-bot.yml)   
[avm.res.insights.action-group](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.action-group.yml)   
[avm.res.insights.activity-log-alert](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.activity-log-alert.yml)
[avm.res.insights.component](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.component.yml)
[avm.res.insights.data-collection-endpoint](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.data-collection-endpoint.yml)
[avm.res.insights.data-collection-rule](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.data-collection-rule.yml)
[avm.res.insights.diagnostic-setting](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.diagnostic-setting.yml)
[avm.res.insights.metric-alert](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.metric-alert.yml)   
[avm.res.insights.private-link-scope](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.private-link-scope.yml)
[avm.res.insights.scheduled-query-rule](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.scheduled-query-rule.yml)
[avm.res.insights.webtest](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.insights.webtest.yml)
[avm.res.key-vault.vault](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.key-vault.vault.yml)
[avm.res.kubernetes-configuration.extension](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.kubernetes-configuration.extension.yml)
[avm.res.kubernetes-configuration.flux-configuration](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.kubernetes-configuration.flux-configuration.yml)
[avm.res.logic.workflow](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.logic.workflow.yml)
[avm.res.machine-learning-services.workspace](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.machine-learning-services.workspace.yml)
[avm.res.maintenance.maintenance-configuration](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.maintenance.maintenance-configuration.yml)
[avm.res.managed-identity.user-assigned-identity](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.managed-identity.user-assigned-identity.yml)
[avm.res.management.management-group](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.management.management-group.yml)
[avm.res.net-app.net-app-account](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.net-app.net-app-account.yml)
[avm.res.network.application-security-group](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.application-security-group.yml)
[avm.res.network.azure-firewall](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.azure-firewall.yml)
[avm.res.network.bastion-host](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.bastion-host.yml)      
[avm.res.network.connection](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.connection.yml)
[avm.res.network.ddos-protection-plan](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.ddos-protection-plan.yml)
[avm.res.network.dns-forwarding-ruleset](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.dns-forwarding-ruleset.yml)
[avm.res.network.dns-resolver](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.dns-resolver.yml)      
[avm.res.network.dns-zone](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.dns-zone.yml)
[avm.res.network.express-route-circuit](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.express-route-circuit.yml)
[avm.res.network.express-route-gateway](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.express-route-gateway.yml)
[avm.res.network.firewall-policy](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.firewall-policy.yml)
[avm.res.network.front-door-web-application-firewall-policy](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.front-door-web-application-firewall-policy.yml)
[avm.res.network.front-door](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.front-door.yml)
[avm.res.network.ip-group](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.ip-group.yml)
[avm.res.network.load-balancer](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.load-balancer.yml)   
[avm.res.network.local-network-gateway](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.local-network-gateway.yml)
[avm.res.network.nat-gateways](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.nat-gateway.yml)        
[avm.res.network.network-interface](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.network-interface.yml)
[avm.res.network.network-manager](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.network-manager.yml)
[avm.res.network.network-security-group](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.network-security-group.yml)
[avm.res.network.private-dns-zone](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.private-dns-zone.yml)
[avm.res.network.private-endpoint](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.private-endpoint.yml)
[avm.res.network.private-link-service](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.private-link-service.yml)
[avm.res.network.public-ip-address](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.public-ip-address.yml)
[avm.res.network.public-ip-prefix](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.public-ip-prefix.yml)
[avm.res.network.route-table](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.route-table.yml)
[avm.res.network.trafficmanagerprofile](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.trafficmanagerprofile.yml)
[avm.res.network.virtual-hub](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.virtual-hub.yml)
[avm.res.network.virtual-network-gateway](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.virtual-network-gateway.yml)
[avm.res.network.virtual-network](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.virtual-network.yml)
[avm.res.network.virtual-wan](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.virtual-wan.yml)
[avm.res.network.vpn-gateway](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.vpn-gateway.yml)
[avm.res.network.vpn-site](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.network.vpn-site.yml)
[avm.res.operational-insights.workspace](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.operational-insights.workspace.yml)
[avm.res.operations-management.solution](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.operations-management.solution.yml)
[avm.res.power-bi-dedicated.capacity](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.power-bi-dedicated.capacity.yml)
[avm.res.purview.account](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.purview.account.yml)
[avm.res.resource-graph.query](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.resource-graph.query.yml)      
[avm.res.resources.deployment-script](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.resources.deployment-script.yml)
[avm.res.resources.resource-group](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.resources.resource-group.yml)
[avm.res.search.search-service](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.search.search-service.yml)   
[avm.res.service-bus.namespace](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.service-bus.namespace.yml)   
[avm.res.signal-r-service.signal-r](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.signal-r-service.signal-r.yml)
[avm.res.signal-r-service.web-pub-sub](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.signal-r-service.web-pub-sub.yml)
[avm.res.sql.server](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.sql.server.yml)
[avm.res.storage.storage-account](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.storage.storage-account.yml)
[avm.res.synapse.private-link-hub](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.synapse.private-link-hub.yml)
[avm.res.synapse.workspace](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.synapse.workspace.yml)
[avm.res.virtual-machine-images.image-template](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.virtual-machine-images.image-template.yml)
[avm.res.web.serverfarm](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.web.serverfarm.yml)
[avm.res.web.site](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.web.site.yml)
[avm.res.web.static-site](https://github.com/jtracey93/bicep-registry-modules/actions/workflows/avm.res.web.static-site.yml)     |

## Type of Change

- [x] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: .github/workflows/avm.template.module.yml [hunk 1]
```diff
@@ -81,6 +104,7 @@ jobs:
       (fromJson(inputs.workflowInput)).deploymentValidation == 'true' && needs.job_module_static_validation.result != 'failure'
```
Comment (eriqua) [inline .github/workflows/avm.template.module.yml:104]: We should block next jobs on psrule reliability failures
```suggestion
      (fromJson(inputs.workflowInput)).deploymentValidation == 'true' && needs.job_module_static_validation.result != 'failure' && needs.job_psrule_test_waf_reliability != 'failure'
```

File: .github/actions/templates/avm-validateModulePSRule/action.yml [hunk 2]
```diff
@@ -37,6 +37,10 @@ inputs:
   psrulePath:
     description: "The path to PSRule configurations"
     required: false
+  psruleBaseline:
```
Comment (eriqua) [inline .github/actions/templates/avm-validateModulePSRule/action.yml:40]: Once the PSRule feature becomes available, unblocking this PR, I'd suggest to leverage the avm/utilities/tools/Invoke-WorkflowsForBranch.ps1 utility to trigger all module workflows. Enabling only static validation is enough. The utility will provide workflow badges as an output, that we can copy in the PR body to verify the reliability compliance for the whole library.

Comment (eriqua) [reply .github/actions/templates/avm-validateModulePSRule/action.yml:40]: Unresolved to get badges attached before the MR is merged

Comment (jtracey93) [reply .github/actions/templates/avm-validateModulePSRule/action.yml:40]: done

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**